weather/www/js/services.js

```diff
@@ -0,0 +1,214 @@
+        };
+
+        // Round temp to tenths of a degree.
+        this.roundTemp = function(temp) {
+            if (Math.abs(temp) >= 10) {
+                return temp.toPrecision(3);
+            } else {
+                return temp.toPrecision(2);
+            }
+        };
+
+        // Return current temperature
+        this.tempNow = function() {
+            return this.roundTemp(weatherStore.current.currently.temperature);
+        };
+
+        // Return weather data for tomorrow.
+        this.weatherTomorrow = function() {
+			var temp = this.roundTemp(
+				weatherStore.current.daily.data[1].temperatureMax);
+			var precip = weatherStore.current.daily.data[1].precipIntensity;
+			return [precip, temp,];
+        };
+		
+		// Return weather data for day 2
+        this.weatherDay2 = function() {
+            var temp = this.roundTemp(
+						weatherStore.current.daily.data[2].temperatureMax);
+			var precip = weatherStore.current.daily.data[2].precipIntensity;
+			return [precip, temp];
+        };
+		
+		// Return weather data for day 3
+        this.weatherDay3 = function() {
+            var temp = this.roundTemp(
+						weatherStore.current.daily.data[3].temperatureMax);
+			var precip = weatherStore.current.daily.data[3].precipIntensity;
+			return [precip, temp];
+        };
+			
+		// Return weather data for day 4
+        this.weatherDay4 = function() {
+            var temp = this.roundTemp(
+						weatherStore.current.daily.data[4].temperatureMax);
+			var precip = weatherStore.current.daily.data[4].precipIntensity;
+			return [precip, temp];
+        };
+		
+		// Return weather data for day 5
+        this.weatherDay5 = function() {
+            var temp = this.roundTemp(
+						weatherStore.current.daily.data[5].temperatureMax);
+			var precip = weatherStore.current.daily.data[5].precipIntensity;
+			return [precip, temp];
+        };
+		
+		// Return weather data for day 6
+        this.weatherDay6 = function() {
+            var temp = this.roundTemp(
+						weatherStore.current.daily.data[6].temperatureMax);
+			var precip = weatherStore.current.daily.data[6].precipIntensity;
+			return [precip, temp];
+        };
+		
+		//Return weather for tonight
+        this.tempToMidnightLow = function() {
+            var low = this.tempNow();
+            var start = this.findHourNow();
+            var end = this.findHourMidnight();
+            if (start >= 0 && end >= 0) {
+                for (var i = start; i <= end; i++) {
+                    low = Math.min(low,
+                            weatherStore.current.hourly.data[i].temperature);
+                }
+            }
+
+            return this.roundTemp(low);
+        };
+		
+		//Return the intensity of precipitation
+		this.todayPrecip = function() {
+			return weatherStore.current.hourly.data[0].precipIntensity;
+		};
+		
+		//Return the correct picture for the current day
+		//depending on the current weather
+		this.getPicture = function() {
```

Comment:
I didn't, but I think it would actually be better to use the ionicicons rather than using images hosted on the web like i did, so I plan on doing just that.